### DISCUSSION TITLE
Help with Generating and Saving a Sketch/Face

### DISCUSSION THREAD

**Discussion by rdaigle00** [ansys/pyansys-geometry#914] [Q&A]
Hi, 

I have been following through the examples and API reference documentation. I am trying to create a 2D sketch with a complex contour to use in a 2D PyFluent simulation. I have been able to successfully generate and visualize my intended sketch. 

I'm currently unsure of how to generate a 2D surface/face and more importantly how to properly save/export the file to be used to generate a mesh (either in PyFluent or PyPrimeMesh). I would be very grateful for any advice or being pointed to any documentation that might assist me.

Thanks!

**Comment by RobPasMue** [comment]
Hi @rdaigle00!

The current ``Sketch`` capabilities are a client-side only feature for PyAnsys Geometry. Also, they are purely in memory for now, meaning that you cannot perform a ``Sketch`` export to a file. However, I do believe that this request is really interesting and I'd be glad to support its implementation. There are a couple of things though that would be interesting to know.

* Are you only interested in 2D or would you be able to perform your 3D simulation with an "infinitely" long extrusion? If this is the case, then you can extrude your sketch by an "infinite" distance and export the ``Design`` into one of the supported formats. PyPrimeMesh is capable of reading FMD easily, for example.
* Do you have an Ansys CAD product installed in your machine (Discovery, SpaceClaim, Geometry Service)? If not, the export of a ``Sketch`` into a file that other Ansys products read might be more difficult.

In the meantime, I will also double check with the CAD team to see how we can ease the export of sketches to other products.

Thanks for your question @rdaigle00!

**Reply by RobPasMue** [reply]
Pinging @ansys/pyansys-geometry-maintainers for visibility

**Reply by rdaigle00** [reply]
Hi @RobPasMue thank you for the quick response! 

- For this project, I'm only interested in a 2D geometry/mesh since I'm modeling an axisymmetric device (effectively a pipe with a changing cross-sectional area and an annulus).
- I have SpaceClaim and Geometry Service both installed on my machine as well as ANSYS Mechanical and Fluent.

To give some additional context, this project is something I was working on through Workbench using Design Modeler, ANSYS Meshing , and Fluent. As part of this project I need to analyze an array of different designs across both different fluid conditions and geometries. I was hoping to transition to using the PyANSYS interface to make this a bit easier to connect directly with like a CSV of design points.
